packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1496,35 +1497,43 @@ export default abstract class SqlIntegration
       name: string;
       type: MetricType;
       sql: string;
+      exists?: boolean;
     }[] = [];
 
+    const binomialSqlQuery = this.getAutoGeneratedMetricSqlQuery(
+      result.event,
+      result.hasUserId,
+      schemaFormat,
+      "binomial"
+    );
+
     //TODO Build some logic where based on the event, we determine what metrics to create (by default, we create binomial and count) for every event
     metricsToCreate.push({
       name: result.displayName,
       type: "binomial",
-      sql: this.getAutoGeneratedMetricSqlQuery(
-        result.event,
-        result.hasUserId,
-        schemaFormat,
-        "binomial"
-      ),
+      exists: existingMetrics.some((metric) => metric.sql === binomialSqlQuery),
```

Comment:
When checking for an existing query, I would also check the `type`.  There may be a couple other fields we also want to compare, so abstracting this check to a function could be good.